bindings/python/src/trainers.rs

```diff
@@ -41,6 +45,34 @@ impl PyTrainer {
         })
     }
 }
+#[pymethods]
+impl PyTrainer {
+    fn __getstate__(&self, py: Python) -> PyResult<PyObject> {
+        let data = serde_json::to_string(&self.trainer).map_err(|e| {
```

Comment:
`serde_json` ? Shouldn't you be using [serde-pickle](https://docs.rs/serde-pickle/latest/serde_pickle/) ?

Reply:
Well we're in control of the bytes here, so we don't need to use pickle.

It's more consistent with the rest of the lib too (Moving would be another PR imo).

Reply:
Would I be able to unpickle the output from python code without tokenizers imported ? I don't see any uses of having proper pickling, it's just to understand what's happening here :)

Reply:
Also I feel the need to mention that the json pickle thing is already being relied on : https://github.com/huggingface/tokenizers/issues/947 :).

(I think we should make the things more printable in python itself without relying on this, but hey.)